powersimdata/input/tests/test_change_table.py

```diff
@@ -492,3 +510,66 @@ def test_remove_bus(ct):
         ct.remove_bus({845})
     ct.scale_plant_capacity(resource="ng", plant_id={0: 0})
     ct.remove_bus({845})
+
+
+def test_add_demand_flexibility(ct):
+    with pytest.raises(Exception):
+        # Fails because "demand_flexibility_dn", a required key, is not included
+        ct.add_demand_flexibility(
+            {"demand_flexibility_up": "Test", "demand_flexibility_duration": 6}
+        )
+
+    with pytest.raises(Exception):
+        # Fails because there is a key that should not be there
+        ct.add_demand_flexibility(
+            {
+                "demand_flexibility_up": "Test",
+                "demand_flexibility_dn": "Test",
+                "demand_flexibility_duration": 6,
+                "demand_flexibility_wrong_key": "Test",
+            }
+        )
+
+    with pytest.raises(Exception):
+        # Fails because there are no profiles available that match the specified version
+        ct.add_demand_flexibility(
+            {
+                "demand_flexibility_up": "Test",
+                "demand_flexibility_dn": "Test",
+                "demand_flexibility_duration": 6,
+            }
+        )
+
+    # Create fake files in the expected directory path
+    exp_path = os.path.join(Path.home(), "ScenarioData", "raw", str(grid.grid_model))
+    dir_exists_prev = True
+    if not os.path.isdir(exp_path):
+        os.makedirs(exp_path)
+        dir_exists_prev = False
+    fake_df = pd.DataFrame()
+    fake_df.to_csv(os.path.join(exp_path, "demand_flexibility_up_Test.csv"))
+    fake_df.to_csv(os.path.join(exp_path, "demand_flexibility_dn_Test.csv"))
+
+    # Add a test instance of demand flexibility to the change table
+    ct.add_demand_flexibility(
+        {
+            "demand_flexibility_up": "Test",
+            "demand_flexibility_dn": "Test",
+            "demand_flexibility_duration": 6,
+        }
+    )
+    exp_dict = {
+        "demand_flexibility": {
+            "demand_flexibility_up": "Test",
+            "demand_flexibility_dn": "Test",
+            "demand_flexibility_duration": 6,
+        },
+    }
+    assert ct.ct == exp_dict
+
```

Comment:
I know this is probably safe, but also think we could use a temporary folder instead. Commenting to remind myself to do that later

Reply:
Do you mean instead of having it in the `ScenarioData` folder? If so, I originally tried to do that, but `get_profile_version_local` looks for the local directory, which is set to a `ScenarioData` folder in the user's home directory.

Reply:
Yeah, it wasn't trivial to do but I pushed a branch [here](https://github.com/Breakthrough-Energy/PowerSimData/compare/lane/demand_flex...jon/demand_flex?expand=1) that I think would work. The main aspect is that we mock the `get_data_access` method so that it returns an instance which uses a `TempFS` instead of the local filesystem. If it looks good I can make a PR to your branch.

Reply:
I'm not as well-versed with the `fs` package or with some of the data access functionality, but it looks good to me. The only other thing I can see is that a similar change would need to be made to the test I added in `test_change_table.py` to get rid of some of the similar file and directory creation.

Reply:
Ah thanks, I missed that one, will do.